Airwallex/Payment/Sources/PaymentMethodProvider.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  PaymentMethodProvider.swift
+//  Airwallex
+//
+//  Created by Weiping Li on 2024/12/12.
+//  Copyright © 2024 Airwallex. All rights reserved.
+//
+
+import Foundation
+
+@MainActor
+final class PaymentMethodProvider {
+    private let provider: AWXPaymentMethodListViewModel
+    enum Foo: String {
+        case one
+    }
+    var session: AWXSession {
+        provider.session
+    }
+    
+    private(set) var methods = [AWXPaymentMethodType]()
+    private(set) var consents = [AWXPaymentConsent]()
+    
+    init(provider: AWXPaymentMethodListViewModel) {
+        self.provider = provider
+    }
+    
+    var isApplePayAvailable: Bool {
+        guard let applePay = methods.first(where: { $0.name == AWXApplePayKey }) else {
+            return false
+        }
+        return true
+    }
+    
+    func fetchPaymentMethods() async throws {
+        let (methods, consents) = try await provider.fetchAvailablePaymentMethodsAndConsents()
+        // even if there is no paymmentMethods defined in session, we still need to
+        // make sure the payment methods and consents are unique
+        let availableMethods = session.filteredPaymentMethodTypes(methods)
+        var filteredConsents = [AWXPaymentConsent]()
+        var methodDict = Dictionary(
+            uniqueKeysWithValues: zip(
+                availableMethods.map { $0.name.lowercased() },
+                availableMethods
+            )
+        )
+        // filter methods
+        var set = Set<String>()
+        var filteredMethods = [AWXPaymentMethodType]()
+        if let predefinedMethods = session.paymentMethods, !predefinedMethods.isEmpty {
+            for predefined in predefinedMethods.map({ $0.lowercased() }) {
+                if let method = methodDict[predefined], !set.contains(predefined) {
+                    filteredMethods.append(method)
+                    set.insert(predefined)
```

Comment:
https://developer.apple.com/documentation/swift/set/insert(_:)-nads `Set` won't allow duplicate element, means even we insert a duplicate element, it won't exist in the new set, so I guess we don't even need to check `set.contains()`?